cpp/src/plasma/CMakeLists.txt

```diff
@@ -126,10 +126,24 @@ if ("${COMPILER_FAMILY}" STREQUAL "gcc")
     " -Wno-conversion")
 endif()
 
+list(APPEND PLASMA_EXTERNAL_STORE_SOURCES "external_store_worker.cc" "external_store.cc" "hash_table_store.cc")
+
+if (PLASMA_EXTRA_EXTERNAL_STORE_SOURCES)
```

Comment:
This seems in such a fashion that you would need to checkout Arrow sources and place your custom source files in it to extend the stores. This won't be a viable way to make this freely extendables.

Reply:
You can specify external sources with -DPLASMA_EXTERNAL_STORE_SOURCES in arrow's cmake build, and don't have to place them in Arrow sources. I am open to suggestions if there is a better alternative.

Reply:
Why do you need this at all?

Reply:
So that users can add implementations for new external stores without having to add them to the Arrow codebase.

Reply:
This is actually something I don't want to support in the Arrow codebase. As you already have a register mechanism, users should use that instead of trying to build their own Arrow with their custom sources added.

Reply:
My register mechanism still requires the implementation to be compiled into the `plasma_store_server` executable, as the register happens during static initialization. I don't see how we can perform registration of custom external sources during runtime, and avoid compiling the sources with the `plasma_store_server` altogether..